packages/core/src/App/Containers/Routes/routes.jsx

```diff
@@ -18,65 +18,66 @@ const Error = Loadable({
     },
 });
 
-class Routes extends React.Component {
-    unlisten_to_change = null;
-    initial_route = null;
+const Routes = ({ error, has_error, history, is_logged_in, location, passthrough, ...props }) => {
```

Comment:
we can remove ... props and add addRouteHistoryItem, setInitialRouteHistoryItem instead and remove all props.

Reply:
When I do that, I get an error about `this` in `common-store` is empty, do you have any idea how to solve it?
![Screen Shot 2020-12-13 at 10 01 11 AM](https://user-images.githubusercontent.com/68377532/102006508-dec65400-3d36-11eb-9db8-e83066bf3260.png)

Reply:
@Bahar-FS 
I think it's related to setInitialRouteHistoryItem definition.
in the common store, it set as a function not as an action if you add @action.bound for this function I think the problem will solve

Reply:
@mehrzad-fs Great, yeah, problem solved.